### PR TITLE
Add infraction to charge class

### PR DESCRIPTION
**Pull request by colincadams**
In understanding the difference between an infraction and a civil offense I found the following description useful:

> An infraction, sometimes called a petty offense, is the violation of an administrative regulation, an ordinance, a municipal code, and, in some jurisdictions, a state or local traffic rule. In many states an infraction is not considered a criminal offense and thus not punishable by incarceration. Instead, such jurisdictions treat infractions as civil offenses. 

From https://criminal.findlaw.com/criminal-law-basics/infractions.html

### CODE REVIEW

**Comment by colincadams** [comment]
The alembic migration is going to require some testing. I put it together based on various stack overflow answers, but don't know if it will work.

It is possible to add an enum value using `ALTER TYPE enum_type ADD VALUE 'new_value';`, however this has to be done outside of a transaction and there is not a similar command to remove an enum value. Because of this I implemented both by renaming the enum and replacing it with a new one, so that the upgrade and downgrade mirrored each other and I didn't have to figure out how to do this single migration outside of a transaction.

**Comment by terinpw** [comment]
Going to let Zach comment more on the migration, but otherwise this looks good.

**Comment by jessex** [comment]
This seems like a reasonable approach to have a parallel upgrade and downgrade. I'd suggest trying it out in stage before having a go at production, and then merging only once it has worked in stage.

**Comment by zdg2102** [comment]
Nice, thanks for figuring out how to handle this case. I'll update the README to outline this as the process for adding enum values.

**Comment by colincadams** [comment]
Sounds good, @zdg2102 let's try to run the upgrade on dev with and without that line to see if it works and then we can merge!

File: recidiviz/persistence/database/migrations/versions/3bf06134321e_add_chargeclass_infraction.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# Without 'INFRACTION'
+old_enum_values = ['CIVIL', 'EXTERNAL_UNKNOWN', 'FELONY', 'MISDEMEANOR',
+                   'PAROLE_VIOLATION', 'PROBATION_VIOLATION']
+# With 'INFRACTION'
+new_enum_values = ['CIVIL', 'EXTERNAL_UNKNOWN', 'FELONY', 'INFRACTION',
+                   'MISDEMEANOR', 'PAROLE_VIOLATION', 'PROBATION_VIOLATION']
+
+def upgrade():
+    op.execute('ALTER TYPE charge_class RENAME TO charge_class_old;')
+    sa.Enum(*new_enum_values, name='charge_class').create(bind=op.get_bind())
```
[the remark this answers is not in the document]
Comment (zdg2102) [reply recidiviz/persistence/database/migrations/versions/3bf06134321e_add_chargeclass_infraction.py]: Tested offline, removing the explicit enum creation causes the transaction to fail, probably due to a difference between the create and alter commands.

**Pull request merged**